(SERVER-1247) Packaging changes for bootstrap improvements

jpinsonault opened the pull request:
This commit updates ezbake files and settings to take advantage of
ezbake and tk improvements to allow a project to split its
bootstraps into two services.d directories.

To accommodate a new directory of files to be placed under /opt/,
the ezbake/config dir now has two subdirectories, public and private.
Everything under public is the same as it was before, under config/,
with the exception of the new services.d dir, which contains ca.cfg.

The remainder of the bootstrap entries are in
ezbake/config/private/services.d/bootstrap.cfg

The files under /ext/config have been moved to /ext/config/public
to mirror the ezbake folder structure

---------------

I'm just throwing up this PR for people to look at. The basic structure probably won't change too much
This shouldn't be merged yet for a few reasons:
* There needs to be a lein-ezbake release. PR: https://github.com/puppetlabs/ezbake/pull/305
* The names of the new lein settings and the new directories will be changing based on the above ezbake PR

cprice404 commented on ext/config/public/services.d/bootstrap.cfg:
```diff
@@ -0,0 +1,5 @@
+puppetlabs.services.ca.certificate-authority-service/certificate-authority-service
+# To disable the CA service, comment out the above line and uncomment the line below
+#puppetlabs.services.ca.certificate-authority-disabled-service/certificate-authority-disabled-service
+
```
I think we should call this file `ca.cfg`.  Also I think I'd probably vote for using `system` and, maybe, `user` instead of `public` and `private`, as per @nathanielksmith 's suggestion.

jpinsonault replied:
I just noticed this file exists. It's not supposed to be in the PR. There is a ca.cfg below in the ezbake/config/ dir. Must have borked a copy/paste somewhere

cprice404 commented on ezbake/config/private/services.d/bootstrap.cfg:
```diff
@@ -0,0 +1,11 @@
+puppetlabs.services.puppet-admin.puppet-admin-service/puppet-admin-service
+puppetlabs.services.legacy-routes.legacy-routes-service/legacy-routes-service
+puppetlabs.trapperkeeper.services.authorization.authorization-service/authorization-service
+puppetlabs.services.versioned-code-service.versioned-code-service/versioned-code-service
```
Had been thinking that we might call this file `puppetserver.cfg` or `base.cfg` or something, though I don't have a big problem with keeping it as `bootstrap.cfg`.

cprice404 commented on ezbake/config/public/conf.d/auth.conf:
```diff
@@ -0,0 +1,126 @@
+authorization: {
```
Confused as to why github is showing this file above as "renamed without changes", but is also showing the contents here?  :sheep:

cprice404 commented on ezbake/config/public/services.d/ca.cfg:
```diff
@@ -0,0 +1,5 @@
+puppetlabs.services.ca.certificate-authority-service/certificate-authority-service
+# To disable the CA service, comment out the above line and uncomment the line below
+#puppetlabs.services.ca.certificate-authority-disabled-service/certificate-authority-disabled-service
+
```
This is super weird that this is showing up in addition to the `bootstrap.cfg` in the public directory above...

cprice404 commented:
+1 other than waiting for SNAPSHOT releases to come out of the oven, figuring out what is going on with the file renames / duplicates, and fixing the setting names once ezbake is ready.

jpinsonault commented:
Updated with new names, removed the extra files, updated to TK 1.4.0. Only blocker now is the lein-ezbake release after https://github.com/puppetlabs/ezbake/pull/305 is merged

rlinehan commented:
👍 once the ezbake dependency has been updated

jpinsonault commented:
Added a commit to match the folder structure changes in https://github.com/puppetlabs/ezbake/pull/310. Still waiting on ezbake release

jpinsonault commented:
ezbake release is happening soon, but we'll wait to merge this until all the "branchnanigans" are over, as @cprice404 put it

jpinsonault commented:
Updated ezbake to `0.4.0` and rebased

cprice404 commented:
@rlinehan it seems like you and I are both +1, and all of the prereqs for merging this are behind us... so, any objections to going ahead and merging it now?

rlinehan commented:
:shipit:
